### DISCUSSION TITLE
How to generate a image in command line mode?

### DISCUSSION THREAD

**Discussion by inspire-boy** [hlky/stable-diffusion-webui#1349] [General]
How to generate a image in command line mode?
I want to use webui to make a APi,   like this 
`python script/txt2img --prompt "a god in street."`, But some error raised
KeyError: 'st.session_state has no key "defaults". Did you forget to initialize it? More info: https://docs.streamlit.io/library/advanced-features/session-state#initialization'

**Comment by sherry0429** [comment]
this error because streamlit init variables when bootstrap. you run python xxx instead of streamlit run main.py, so st.session_state dict is empty. that is why this exception raised.

you could fix this simply by these steps:
1. edit code, print st.session_state before each line txt2img call it.
2. run webui.cmd, so streamlit will run, choose txt2img and run once.
3. you will see some prints output in your cmd like "{defaults : "xxxx"}", copy them
4. replace value in txt2img, don't use st.session_state, use static string variables
5. run txt2img again.

If this help you, please accept this answer, thank you @inspire-boy
